src/kilo.c

```diff
@@ -1009,13 +1010,60 @@ char *editorReadStringFromStatusBar(char *prefix) {
       break;
     }
 
+
```

Comment:
I commented this earlier, but I can't seem to find it:

I originally intended this function to be more general than colon functions, (ie, used whenever we want to read a string from the user). Can you do history with a callback function pointer that clients can decide to populate or pass NULL in to preserve this behaviour?